upstream bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1657849
# Updated for Firefox/LibreWolf 158: OpenH264 added sched_getaffinity handling
# to the plugin sandbox; preserve the musl sched_setscheduler workaround there too.
diff --git a/security/sandbox/linux/SandboxFilter.cpp b/security/sandbox/linux/SandboxFilter.cpp
--- a/security/sandbox/linux/SandboxFilter.cpp
+++ b/security/sandbox/linux/SandboxFilter.cpp
@@ -1920,13 +1920,20 @@
       case __NR_sched_get_priority_min:
       case __NR_sched_get_priority_max:
         return Allow();
       // The OpenH264 plugin needs sched_getaffinity() in multithreaded mode;
       // bug 2071378.
       case __NR_sched_getaffinity:
       case __NR_sched_getparam:
       case __NR_sched_getscheduler:
-      case __NR_sched_setscheduler: {
+#ifdef __GLIBC__
+      case __NR_sched_setscheduler:
+#endif
+      {
         Arg<pid_t> pid(0);
         return If(pid == 0, Allow()).Else(Trap(SchedTrap, nullptr));
       }
+#ifndef __GLIBC__
+      case __NR_sched_setscheduler:
+        return Allow();
+#endif
 
@@ -2209,13 +2216,21 @@
       case __NR_sched_getparam:
       case __NR_sched_setparam:
       case __NR_sched_getscheduler:
+#ifdef __GLIBC__
       case __NR_sched_setscheduler:
+#endif
       case __NR_sched_getattr:
       case __NR_sched_setattr: {
         Arg<pid_t> pid(0);
         return If(pid == 0, Allow()).Else(Trap(SchedTrap, nullptr));
       }
 
+#ifndef __GLIBC__
+      // sched_setscheduler gets special treatment here (bug 1657849):
+      case __NR_sched_setscheduler:
+        return Allow();
+#endif
+
         // The priority bounds are also used, sometimes (bug 1838675):
       case __NR_sched_get_priority_min:
       case __NR_sched_get_priority_max:
